client/app/services/parameters/TextParameter.js

```diff
@@ -0,0 +1,16 @@
+import { toString, isEmpty } from 'lodash';
+import { Parameter } from '.';
+
+class TextParameter extends Parameter {
+  constructor(parameter, parentQueryId) {
+    super(parameter, parentQueryId);
+    this.setValue(parameter.value);
+  }
+
+  getValue() {
+    const value = toString(this.value);
+    return !isEmpty(value) ? value : null;
+  }
+}
```

Comment:
Perhaps can also have a `normalize()` override which acknowledges 0 for NumberParameter and '' for TextParameter etc.

Reply:
So far it was ok to have the logic in `getValue` as the method was simple. Do you see a case where we will need the `normalize` outside that method?

Reply:
Perhaps the param's `isDirty()` could use it too